packages/semantic-dom-diff/chai-dom-diff.js

```diff
@@ -37,21 +57,31 @@ export const chaiDomDiff = (chai, utils) => {
   const getLightDomHtml = el => el.innerHTML;
   const getShadowDomHtml = el => getCleanedShadowDom(el);
 
-  /** Base HTML assertion for `assert` interface. */
-  const assertHtmlEquals = (actual, expected, options, negate = false) => {
+  /**
+   * Base HTML assertion for `assert` interface.
+   * @param {string | Node} actual
+   * @param {string | Node} expected
+   * @param {boolean} negate
+   * @param {[string]|[DiffOptions]|[string, DiffOptions]} rest
+   */
+  const assertHtmlEquals = (actual, expected, negate, ...rest) => {
+    const { message, options } = disambiguateArgs(...rest);
     // use chai's built-in string comparison, log the updated snapshot on error
-    const assertion = new chai.Assertion(getDiffableHTML(actual, options));
+    const assertion = new chai.Assertion(getDiffableHTML(actual, options), message);
     const expectedDiffableHTML = getDiffableHTML(expected, options);
 
     if (negate) {
-      assertion.not.equal(expectedDiffableHTML);
+      assertion.not.equal(expectedDiffableHTML, message);
     } else {
-      assertion.equal(expectedDiffableHTML);
+      assertion.equal(expectedDiffableHTML, message);
     }
   };
 
   /** DOM assertion for `should` and `expect` interfaces. */
   const domEquals = _super =>
+    /**
+     * @this {any}
```

Comment:
what is `this` here?

Reply:
chai itself... chai works by binding its plugin function to itself... that's also the reasons that `handleDom` can not be an arrow function... as it would break chains plugin "system"...

it's a very old school system... because it's old like really old 😅

Reply:
is that Chai or Chai.ChaiStatic?

Reply:
I think this instance is actually `Chai.AssertionStatic`